modules/margin-protocol/src/lib.rs

```diff
@@ -967,6 +1006,98 @@ impl<T: Trait> Module<T> {
 		Ok(())
 	}
 
+	/// get trader's max threshold based on opened positions after performing an action.
+	///
+	/// Return `RiskThreshold` or `Default` value.
+	fn _get_trader_max_threshold(who: &T::AccountId, action: Action<T>) -> RiskThreshold {
```

Comment:
```suggestion
	fn _risk_threshold_of_trader(who: &T::AccountId) -> RiskThreshold {
```

Let's keep the naming consistent with other functions.

And I would recommend not passing `Action` all along to here, as not all helper functions should understand what `action` mean to them. Keep `action` to `_check_trader` should be enough for all scenarios.

Reply:
If call `_check_trader` directly in _open_position,  will need to match the result.
So, add Action to `_ensure_trader_safe`, and keep the same to `_ensure_pool_safe`

Reply:
I'm not recommending `_check_trader` should be called in `open_position`. `_check_trader` should cover all the scenarios with `action` and don't need to pass along it to `_risk_threshold_of_trader`.

Reply:
risk_threshold_of_trader will change when opening a new trading pair position. So I check in `open_position`

Reply:
`_risk_threshold_of_trader` need new pair information, but the `pair` info in `open_position`. So I pass it through `_ensure_trader_safe`